crates/radix-engine-toolkit/tests/transaction_types.rs

```diff
@@ -2576,3 +2576,90 @@ fn subintent_manifest_classification_test(
         assert!(!contains_general_subintent_classification);
     }
 }
+
+#[test]
+fn update_to_account_owner_keys_is_reserved_instruction() {
+    let account =
+        ComponentAddress::new_or_panic([EntityType::GlobalAccount as u8; 30]);
+    reserved_instructions_test(
+        ManifestBuilder::new()
+            .set_metadata(account, "owner_keys", "not relevant")
+            .build(),
+        true,
+        false,
+    )
+}
+
+#[test]
+fn update_to_identity_owner_keys_is_reserved_instruction() {
+    let identity =
+        ComponentAddress::new_or_panic([EntityType::GlobalIdentity as u8; 30]);
+    reserved_instructions_test(
+        ManifestBuilder::new()
+            .set_metadata(identity, "owner_keys", "not relevant")
+            .build(),
+        false,
+        true,
+    )
+}
+
+#[test]
+fn update_to_component_owner_keys_is_not_reserved_instruction() {
+    let component = ComponentAddress::new_or_panic(
+        [EntityType::GlobalGenericComponent as u8; 30],
+    );
+    reserved_instructions_test(
+        ManifestBuilder::new()
+            .set_metadata(component, "owner_keys", "not relevant")
+            .build(),
+        false,
+        false,
+    )
+}
+
+#[test]
+fn update_to_account_non_owner_keys_field_is_not_reserved_instruction() {
+    let account =
+        ComponentAddress::new_or_panic([EntityType::GlobalAccount as u8; 30]);
+    reserved_instructions_test(
+        ManifestBuilder::new()
+            .set_metadata(account, "example", "not relevant")
+            .build(),
+        false,
+        false,
+    )
+}
+
+#[test]
+fn update_to_identity_non_owner_keys_field_is_not_reserved_instruction() {
+    let identity =
+        ComponentAddress::new_or_panic([EntityType::GlobalIdentity as u8; 30]);
+    reserved_instructions_test(
+        ManifestBuilder::new()
+            .set_metadata(identity, "example", "not relevant")
+            .build(),
+        false,
+        false,
+    )
+}
+
+fn reserved_instructions_test(
```

Comment:
MINOR/OPTIONAL: Ditto, this test helper rather obscures what the tests are actually asserting. It feels like it's optimizing the tests for DRYness rather than clarity/flexibility.